patroni/postgresql.py

```diff
@@ -164,9 +167,42 @@ def query(self, sql, *params):
     def data_directory_empty(self):
         return not os.path.exists(self.data_dir) or os.listdir(self.data_dir) == []
 
+    @staticmethod
+    def initdb_allowed_option(name):
+        allowed_options = set(['auth', 'auth-host', 'auth-local', 'encoding', 'data-checksums',
```

Comment:
Hm... why not just pass the user options directly to initdb and examine the return code in order to find out if it didn't like some of the options. You have to do it anyway (i.e. what if the user supplies a value for the option that does not need one, i.e. --data-checksums=true).